<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>4 Statistical Analysis | IMAP: Integrated Microbiome Analysis Pipelines</title>
<meta name="author" content="Teresia Mrema-Buza">
<meta name="description" content="4.1 LEfSe: Linear discriminant analysis (LDA) effect size Linear discriminant analysis effect size (LEfSe) is a statistical method used for biomarker discovery and identification of differentially...">
<meta name="generator" content="bookdown 0.39 with bs4_book()">
<meta property="og:title" content="4 Statistical Analysis | IMAP: Integrated Microbiome Analysis Pipelines">
<meta property="og:type" content="book">
<meta property="og:url" content="https://tmbuza.github.io/imap-statistical-analysis/imap-part11.html">
<meta property="og:description" content="4.1 LEfSe: Linear discriminant analysis (LDA) effect size Linear discriminant analysis effect size (LEfSe) is a statistical method used for biomarker discovery and identification of differentially...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="4 Statistical Analysis | IMAP: Integrated Microbiome Analysis Pipelines">
<meta name="twitter:description" content="4.1 LEfSe: Linear discriminant analysis (LDA) effect size Linear discriminant analysis effect size (LEfSe) is a statistical method used for biomarker discovery and identification of differentially...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/Sen-0.4.9/font.css" rel="stylesheet">
<link href="https://fonts.googleapis.com/css2?family=DM%20Mono&amp;display=swap" rel="stylesheet">
<link href="libs/Bitter-0.4.9/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.7.0/transition.js"></script><script src="libs/bs3compat-0.7.0/tabs.js"></script><script src="libs/bs3compat-0.7.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="End-to-End Practical User Guides Using Integrated Approaches"><big>IMAP:</big> Integrated Microbiome Analysis Pipelines</a>:
        <small class="text-muted">End-to-End Practical User Guides Using Integrated Approaches</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">IMAP-STATS: Statistical Analysis of Microbiome Data</a></li>
<li class="book-part">STATISTICAL TESTING</li>
<li><a class="" href="microbiome-statistical-analysis-steps.html"><span class="header-section-number">1</span> Microbiome Statistical Analysis Steps</a></li>
<li class="book-part">STATISTICAL ANALYSIS</li>
<li><a class="" href="statistical-significance-analysis.html"><span class="header-section-number">3</span> Statistical Significance Analysis</a></li>
<li class="book-part">IMAP-PART 11</li>
<li><a class="active" href="imap-part11.html"><span class="header-section-number">4</span> Statistical Analysis</a></li>
<li class="book-part">APPENDIX</li>
<li><a class="" href="imap-github-repos.html"><span class="header-section-number">A</span> IMAP GitHub Repos</a></li>
<li><a class="" href="read-qc-rulegraph.html"><span class="header-section-number">B</span> Read QC Rulegraph</a></li>
<li><a class="" href="session-information.html"><span class="header-section-number">C</span> Session Information</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/tmbuza/imap-machine-learning">View book source <i class="fas fa-air-freshener"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="imap-part11" class="section level1" number="4">
<h1>
<span class="header-section-number">4</span> Statistical Analysis<a class="anchor" aria-label="anchor" href="#imap-part11"><i class="fas fa-link"></i></a>
</h1>
<div id="lda-effect-size" class="section level2" number="4.1">
<h2>
<span class="header-section-number">4.1</span> LEfSe: Linear discriminant analysis (LDA) effect size<a class="anchor" aria-label="anchor" href="#lda-effect-size"><i class="fas fa-link"></i></a>
</h2>
<p>Linear discriminant analysis effect size (LEfSe) is a statistical method used for biomarker discovery and identification of differentially abundant features between two or more biological groups.</p>
<ul>
<li>LEfSe combines LDA (Linear Discriminant Analysis) and effect size estimation to identify microbial taxa or features that are most likely to explain differences between groups.</li>
</ul>
<p><strong>LDA plot</strong></p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files2.html">dir.create</a></span><span class="op">(</span><span class="st">"data"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/files2.html">dir.create</a></span><span class="op">(</span><span class="st">"data"</span><span class="op">)</span><span class="op">}</span></span>
<span></span>
<span><span class="co"># source("_microbial_plot.R")</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://haozhu233.github.io/kableExtra/">kableExtra</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">microbial</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Nationality</span></span>
<span><span class="va">lda_nat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/microbial/man/ldamarker.html">ldamarker</a></span><span class="op">(</span><span class="va">ps_raw</span>, group<span class="op">=</span><span class="st">"nationality"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># p &lt;- lda_nat %&gt;%</span></span>
<span><span class="co"># plotLDA(group=c("AAM","AFR"), lda = 4.5) +</span></span>
<span><span class="co">#   labs(fill = NULL)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # BMI</span></span>
<span><span class="co"># lda_bmi &lt;- ldamarker(ps_raw, group="bmi_group")</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># lda_bmi %&gt;%</span></span>
<span><span class="co"># plotLDA(group=c("lean", "overweight"), lda = 4.5) +</span></span>
<span><span class="co">#   labs(fill = NULL)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># lda_bmi %&gt;%</span></span>
<span><span class="co"># plotLDA(group=c("lean", "obese"), lda = 4) +</span></span>
<span><span class="co">#   labs(fill = NULL)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># lda_bmi %&gt;%</span></span>
<span><span class="co"># plotLDA(group=c("overweight", "obese"), lda = 4.5) +</span></span>
<span><span class="co">#   labs(fill = NULL)</span></span></code></pre></div>
</div>
<div id="taxa-prevalence-and-detection" class="section level2" number="4.2">
<h2>
<span class="header-section-number">4.2</span> Taxa Prevalence and Detection<a class="anchor" aria-label="anchor" href="#taxa-prevalence-and-detection"><i class="fas fa-link"></i></a>
</h2>
<div class="infoicon">
<p>
Taxa prevalence refers to the frequency or occurrence of a particular
taxonomic unit (e.g., a species or genus) within a given dataset or set
of samples.
</p>
<ul>
<li>
It quantifies how often a taxon is present or detected across the
samples under study.
</li>
<li>
High prevalence indicates that the taxon is commonly found across
many samples
</li>
<li>
Low prevalence suggests that the taxon is more sporadic or
rare.
</li>
<li>
Analyzing taxa prevalence can provide insights into the distribution
of different microbial taxa.
</li>
</ul>
</div>
<p><strong>Min and max sample and taxa sums</strong></p>
<ul>
<li>These values are useful when setting up some threshholds.</li>
</ul>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217">phyloseq</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"Minimum sample sums:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_sums.html">sample_sums</a></span><span class="op">(</span><span class="va">ps_raw</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, <span class="st">"\n\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code>Minimum sample sums: 1776 </code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"Maximum sample sums:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_sums.html">sample_sums</a></span><span class="op">(</span><span class="va">ps_raw</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, <span class="st">"\n\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code>Maximum sample sums: 28883 </code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"Minimum taxa sums:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_sums.html">taxa_sums</a></span><span class="op">(</span><span class="va">ps_raw</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, <span class="st">"\n\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code>Minimum taxa sums: 0 </code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"Maximum taxa sums:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_sums.html">taxa_sums</a></span><span class="op">(</span><span class="va">ps_raw</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, <span class="st">"\n\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code>Maximum taxa sums: 873314 </code></pre>
<p><strong>Subsetting to core taxa</strong></p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ps_prune_sample</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_samples-methods.html">subset_samples</a></span><span class="op">(</span><span class="va">ps_raw</span>, <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_sums.html">sample_sums</a></span><span class="op">(</span><span class="va">ps_raw</span><span class="op">)</span> <span class="op">&gt;</span><span class="fl">20000</span><span class="op">)</span></span>
<span><span class="va">ps_prune_taxa</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_taxa-methods.html">prune_taxa</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_sums.html">taxa_sums</a></span><span class="op">(</span><span class="va">ps_prune_sample</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="va">ps_prune_sample</span><span class="op">)</span></span>
<span><span class="va">ps_core</span> <span class="op">&lt;-</span> <span class="fu">microbiome</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microbiome/man/core.html">core</a></span><span class="op">(</span><span class="va">ps_prune_taxa</span>, detection  <span class="op">=</span> <span class="fl">0.01</span>, prevalence <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span>
<span><span class="va">ps_core</span></span></code></pre></div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 116 taxa and 18 samples ]
sample_data() Sample Data:       [ 18 samples by 8 sample variables ]
tax_table()   Taxonomy Table:    [ 116 taxa by 3 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 116 tips and 115 internal nodes ]</code></pre>
</div>
<div id="permanova" class="section level2" number="4.3">
<h2>
<span class="header-section-number">4.3</span> PERMANOVA: Permutation Analysis of Variance<a class="anchor" aria-label="anchor" href="#permanova"><i class="fas fa-link"></i></a>
</h2>
<p>Use PERMANOVA to assess the significance of differences between groups based on multivariate data. PERMANOVA is a flexible and robust method that accounts for various data characteristics and can provide valuable insights into the factors driving variation in the data.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Permanova</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://haozhu233.github.io/kableExtra/">kableExtra</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org">magrittr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">ps_prune_sample</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_samples-methods.html">subset_samples</a></span><span class="op">(</span><span class="va">ps_raw</span>, <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_sums.html">sample_sums</a></span><span class="op">(</span><span class="va">ps_raw</span><span class="op">)</span> <span class="op">&gt;</span><span class="fl">15000</span><span class="op">)</span></span>
<span><span class="va">ps_prune_taxa</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_taxa-methods.html">prune_taxa</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_sums.html">taxa_sums</a></span><span class="op">(</span><span class="va">ps_prune_sample</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="va">ps_prune_sample</span><span class="op">)</span></span>
<span><span class="va">ps_core</span> <span class="op">&lt;-</span> <span class="fu">microbiome</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microbiome/man/core.html">core</a></span><span class="op">(</span><span class="va">ps_prune_taxa</span>, detection  <span class="op">=</span> <span class="fl">0.01</span>, prevalence <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span>
<span><span class="va">ps_core</span></span></code></pre></div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 119 taxa and 76 samples ]
sample_data() Sample Data:       [ 76 samples by 8 sample variables ]
tax_table()   Taxonomy Table:    [ 119 taxa by 3 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 119 tips and 118 internal nodes ]</code></pre>
</div>
<div id="population-wise-multivariate-anova" class="section level2" number="4.4">
<h2>
<span class="header-section-number">4.4</span> Population-wise multivariate ANOVA<a class="anchor" aria-label="anchor" href="#population-wise-multivariate-anova"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vegandevs/vegan">vegan</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">microbial</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ps_raw</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span><span class="fu"><a href="https://rdrr.io/pkg/microbial/man/betatest.html">betatest</a></span><span class="op">(</span>group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"nationality"</span>, <span class="st">"bmi_group"</span><span class="op">)</span>, distance <span class="op">=</span> <span class="st">"bray"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://magrittr.tidyverse.org/reference/aliases.html">set_colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="st">"Df"</span>, <span class="st">"SumsOfSqs"</span>, <span class="st">"MeanSqs"</span>, <span class="st">"FModel"</span>, <span class="st">"R2"</span>, <span class="st">"Pvalue"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span>format <span class="op">=</span> <span class="st">"html"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html">kable_styling</a></span><span class="op">(</span>bootstrap_options <span class="op">=</span> <span class="st">"basic"</span>, full_width <span class="op">=</span> <span class="cn">F</span>, position <span class="op">=</span> <span class="st">"float_left"</span><span class="op">)</span>  <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/column_spec.html">column_spec</a></span><span class="op">(</span><span class="fl">7</span>, bold <span class="op">=</span> <span class="cn">TRUE</span>, color <span class="op">=</span> <span class="st">"black"</span>, background <span class="op">=</span> <span class="st">"#eeeeee"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-table"><table class="table" style="width: auto !important; float: left; margin-right: 10px;">
<thead><tr>
<th style="text-align:left;">
Group
</th>
<th style="text-align:right;">
Df
</th>
<th style="text-align:right;">
SumsOfSqs
</th>
<th style="text-align:right;">
MeanSqs
</th>
<th style="text-align:right;">
FModel
</th>
<th style="text-align:right;">
R2
</th>
<th style="text-align:right;">
Pvalue
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
bmi_group
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2.303012
</td>
<td style="text-align:right;">
1.1515062
</td>
<td style="text-align:right;">
6.793851
</td>
<td style="text-align:right;">
0.0584197
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
0.001
</td>
</tr>
<tr>
<td style="text-align:left;">
bmi_group
</td>
<td style="text-align:right;">
219
</td>
<td style="text-align:right;">
37.118834
</td>
<td style="text-align:right;">
0.1694924
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.9415803
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
bmi_group
</td>
<td style="text-align:right;">
221
</td>
<td style="text-align:right;">
39.421847
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
nationality
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
6.940786
</td>
<td style="text-align:right;">
6.9407856
</td>
<td style="text-align:right;">
47.011174
</td>
<td style="text-align:right;">
0.1760644
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
0.001
</td>
</tr>
<tr>
<td style="text-align:left;">
nationality
</td>
<td style="text-align:right;">
220
</td>
<td style="text-align:right;">
32.481061
</td>
<td style="text-align:right;">
0.1476412
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.8239356
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
nationality
</td>
<td style="text-align:right;">
221
</td>
<td style="text-align:right;">
39.421847
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
NA
</td>
</tr>
</tbody>
</table></div>
</div>
<div id="core-taxa-multivariate-anova" class="section level2" number="4.5">
<h2>
<span class="header-section-number">4.5</span> Core taxa multivariate ANOVA<a class="anchor" aria-label="anchor" href="#core-taxa-multivariate-anova"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ps_core</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span><span class="fu"><a href="https://rdrr.io/pkg/microbial/man/betatest.html">betatest</a></span><span class="op">(</span>group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"nationality"</span>, <span class="st">"bmi_group"</span><span class="op">)</span>, distance <span class="op">=</span> <span class="st">"bray"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://magrittr.tidyverse.org/reference/aliases.html">set_colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="st">"Df"</span>, <span class="st">"SumsOfSqs"</span>, <span class="st">"MeanSqs"</span>, <span class="st">"FModel"</span>, <span class="st">"R2"</span>, <span class="st">"Pvalue"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span>format <span class="op">=</span> <span class="st">"html"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html">kable_styling</a></span><span class="op">(</span>bootstrap_options <span class="op">=</span> <span class="st">"basic"</span>, full_width <span class="op">=</span> <span class="cn">F</span>, position <span class="op">=</span> <span class="st">"float_left"</span><span class="op">)</span>  <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/column_spec.html">column_spec</a></span><span class="op">(</span><span class="fl">7</span>, bold <span class="op">=</span> <span class="cn">TRUE</span>, color <span class="op">=</span> <span class="st">"black"</span>, background <span class="op">=</span> <span class="st">"#eeeeee"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-table"><table class="table" style="width: auto !important; float: left; margin-right: 10px;">
<thead><tr>
<th style="text-align:left;">
Group
</th>
<th style="text-align:right;">
Df
</th>
<th style="text-align:right;">
SumsOfSqs
</th>
<th style="text-align:right;">
MeanSqs
</th>
<th style="text-align:right;">
FModel
</th>
<th style="text-align:right;">
R2
</th>
<th style="text-align:right;">
Pvalue
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
bmi_group
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
0.3580195
</td>
<td style="text-align:right;">
0.1790097
</td>
<td style="text-align:right;">
1.405949
</td>
<td style="text-align:right;">
0.0370904
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
0.196
</td>
</tr>
<tr>
<td style="text-align:left;">
bmi_group
</td>
<td style="text-align:right;">
73
</td>
<td style="text-align:right;">
9.2945867
</td>
<td style="text-align:right;">
0.1273231
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.9629096
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
bmi_group
</td>
<td style="text-align:right;">
75
</td>
<td style="text-align:right;">
9.6526062
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
nationality
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1.4483886
</td>
<td style="text-align:right;">
1.4483886
</td>
<td style="text-align:right;">
13.064104
</td>
<td style="text-align:right;">
0.1500516
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
0.001
</td>
</tr>
<tr>
<td style="text-align:left;">
nationality
</td>
<td style="text-align:right;">
74
</td>
<td style="text-align:right;">
8.2042176
</td>
<td style="text-align:right;">
0.1108678
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.8499484
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
nationality
</td>
<td style="text-align:right;">
75
</td>
<td style="text-align:right;">
9.6526062
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
NA
</td>
</tr>
</tbody>
</table></div>
</div>
<div id="diff_abund" class="section level2" number="4.6">
<h2>
<span class="header-section-number">4.6</span> Differential abundance Analysis<a class="anchor" aria-label="anchor" href="#diff_abund"><i class="fas fa-link"></i></a>
</h2>
<p>Perform differential abundance analysis to identify features that show statistically significant differences in abundance between groups.</p>
<ul>
<li>This analysis allow researchers to determine which features are associated with the studied conditions or factors.</li>
<li>Statistical tests, such as t-tests, Wilcoxon rank-sum tests, or analysis of variance (ANOVA) can help in assessing the significance of the differences and determine if they are beyond what would be expected by chance.</li>
</ul>
<div id="testing-with-deseq2-tool" class="section level3" number="4.6.1">
<h3>
<span class="header-section-number">4.6.1</span> Testing with DESeq2 tool<a class="anchor" aria-label="anchor" href="#testing-with-deseq2-tool"><i class="fas fa-link"></i></a>
</h3>
</div>
</div>
<div id="biomarker-discovery" class="section level2" number="4.7">
<h2>
<span class="header-section-number">4.7</span> Biomarker Discovery<a class="anchor" aria-label="anchor" href="#biomarker-discovery"><i class="fas fa-link"></i></a>
</h2>
<p>Biomarker discovery aims to identify distinct markers or indicators that offer valuable insights into health and disease. These identified biomarkers provide crucial information regarding various aspects. Employing machine learning techniques, such as <code>random forest classification</code>, can further enhance the understanding by providing more in-depth insights, including:</p>
<ul>
<li>Facilitating disease diagnosis</li>
<li>Monitoring and evaluating treatment responses</li>
<li>Improving comprehension of underlying biological processes</li>
<li>Enabling the prediction of outcomes.</li>
</ul>
<div id="microbial-pkg-markers" class="section level3" number="4.7.1">
<h3>
<span class="header-section-number">4.7.1</span> Biomarkers using microbial package<a class="anchor" aria-label="anchor" href="#microbial-pkg-markers"><i class="fas fa-link"></i></a>
</h3>
<p>Microbial R package performms Random Forest Classification then computes the Confusion Matrix as a means of evaluating the modelâ€™s performance.</p>
<blockquote>
<p>More on biomarker discovery techniques in the next sections underMicrobial Machine Learning.</p>
</blockquote>

</div>
</div>
</div>



  <div class="chapter-nav">
<div class="prev"><a href="statistical-significance-analysis.html"><span class="header-section-number">3</span> Statistical Significance Analysis</a></div>
<div class="next"><a href="imap-github-repos.html"><span class="header-section-number">A</span> IMAP GitHub Repos</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#imap-part11"><span class="header-section-number">4</span> Statistical Analysis</a></li>
<li><a class="nav-link" href="#lda-effect-size"><span class="header-section-number">4.1</span> LEfSe: Linear discriminant analysis (LDA) effect size</a></li>
<li><a class="nav-link" href="#taxa-prevalence-and-detection"><span class="header-section-number">4.2</span> Taxa Prevalence and Detection</a></li>
<li><a class="nav-link" href="#permanova"><span class="header-section-number">4.3</span> PERMANOVA: Permutation Analysis of Variance</a></li>
<li><a class="nav-link" href="#population-wise-multivariate-anova"><span class="header-section-number">4.4</span> Population-wise multivariate ANOVA</a></li>
<li><a class="nav-link" href="#core-taxa-multivariate-anova"><span class="header-section-number">4.5</span> Core taxa multivariate ANOVA</a></li>
<li>
<a class="nav-link" href="#diff_abund"><span class="header-section-number">4.6</span> Differential abundance Analysis</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#testing-with-deseq2-tool"><span class="header-section-number">4.6.1</span> Testing with DESeq2 tool</a></li></ul>
</li>
<li>
<a class="nav-link" href="#biomarker-discovery"><span class="header-section-number">4.7</span> Biomarker Discovery</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#microbial-pkg-markers"><span class="header-section-number">4.7.1</span> Biomarkers using microbial package</a></li></ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/tmbuza/imap-machine-learning/blob/gh-pages/01_statistical_analysis.Rmd">View source <i class="fas fa-air-freshener"></i></a></li>
          <li><a id="book-edit" href="https://github.com/tmbuza/imap-machine-learning/edit/gh-pages/01_statistical_analysis.Rmd">Edit this page <i class="fas fa-air-freshener"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container">
    <div class="row">
      <div class="col-12 col-md-6 mt-3">
        <p><strong><u><a class="text-light" href="https://microbiome.complexdatainsights.com"><big>IMAP:</big> Integrated Microbiome Analysis Pipelines</a></u></strong>: End-to-End Practical User Guides Using Integrated Approaches</p>
<br><p>Authored by Teresia Mrema-Buza, Last updated: <b>2024-04-18</b>.</p>
<br>
</div>
      <div class="col-12 col-md-6 mt-3">
        <p>Note: This practical guide was created using the <a class="text-light" href="https://bookdown.org/yihui/bookdown/">bookdown</a> R package, styled with the modern bs4_book format.</p>
      </div>

    </div>
  </div>
</footer>
</body>
</html>
