<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>3 Statistical Significance Analysis | IMAP: Integrated Microbiome Analysis Pipelines</title>
<meta name="author" content="Teresia Mrema-Buza">
<meta name="description" content="This section delves into statistical analysis methods tailored specifically for microbiome data. Statistical analysis plays a crucial role in understanding the complex relationships and patterns...">
<meta name="generator" content="bookdown 0.39 with bs4_book()">
<meta property="og:title" content="3 Statistical Significance Analysis | IMAP: Integrated Microbiome Analysis Pipelines">
<meta property="og:type" content="book">
<meta property="og:url" content="https://tmbuza.github.io/imap-statistical-analysis/statistical-significance-analysis.html">
<meta property="og:description" content="This section delves into statistical analysis methods tailored specifically for microbiome data. Statistical analysis plays a crucial role in understanding the complex relationships and patterns...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="3 Statistical Significance Analysis | IMAP: Integrated Microbiome Analysis Pipelines">
<meta name="twitter:description" content="This section delves into statistical analysis methods tailored specifically for microbiome data. Statistical analysis plays a crucial role in understanding the complex relationships and patterns...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/Sen-0.4.9/font.css" rel="stylesheet">
<link href="https://fonts.googleapis.com/css2?family=DM%20Mono&amp;display=swap" rel="stylesheet">
<link href="libs/Bitter-0.4.9/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.7.0/transition.js"></script><script src="libs/bs3compat-0.7.0/tabs.js"></script><script src="libs/bs3compat-0.7.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="End-to-End Practical User Guides Using Integrated Approaches"><big>IMAP:</big> Integrated Microbiome Analysis Pipelines</a>:
        <small class="text-muted">End-to-End Practical User Guides Using Integrated Approaches</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">IMAP-STATS: Statistical Analysis of Microbiome Data</a></li>
<li class="book-part">STATISTICAL TESTING</li>
<li><a class="" href="microbiome-statistical-analysis-steps.html"><span class="header-section-number">1</span> Microbiome Statistical Analysis Steps</a></li>
<li class="book-part">STATISTICAL ANALYSIS</li>
<li><a class="active" href="statistical-significance-analysis.html"><span class="header-section-number">3</span> Statistical Significance Analysis</a></li>
<li class="book-part">APPENDIX</li>
<li><a class="" href="imap-github-repos.html"><span class="header-section-number">A</span> IMAP GitHub Repos</a></li>
<li><a class="" href="read-qc-rulegraph.html"><span class="header-section-number">B</span> Read QC Rulegraph</a></li>
<li><a class="" href="session-information.html"><span class="header-section-number">C</span> Session Information</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/tmbuza/imap-machine-learning">View book source <i class="fas fa-air-freshener"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="statistical-significance-analysis" class="section level1" number="3">
<h1>
<span class="header-section-number">3</span> Statistical Significance Analysis<a class="anchor" aria-label="anchor" href="#statistical-significance-analysis"><i class="fas fa-link"></i></a>
</h1>
<p>This section delves into statistical analysis methods tailored specifically for microbiome data. Statistical analysis plays a crucial role in understanding the complex relationships and patterns within microbiome datasets, helping researchers uncover significant findings and insights into microbial community dynamics, composition, and responses to various environmental factors or treatments.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span></span>
<span>  echo  <span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>  message  <span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>  warning  <span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>  cache  <span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>  fig.path <span class="op">=</span> <span class="st">"figures"</span>,</span>
<span>  comment  <span class="op">=</span><span class="cn">NA</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load required packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217">phyloseq</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">microbial</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://microbiome.github.io/microbiome">microbiome</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://david-barnett.github.io/microViz">microViz</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\nSaved RData objects\n\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code>
Saved RData objects</code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"../imap-data-exploration/data/dataframe_objects.rda"</span>, verbose <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<pre><code>Loading objects:
  df_GlobalPatterns
  df_dietswap
  df_caporaso
  df_kostic_crc</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"../imap-data-exploration/data/phyloseq_objects.rda"</span>, verbose <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<pre><code>Loading objects:
  ps_GlobalPatterns
  ps_dietswap
  ps_caporaso
  ps_kostic_crc</code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"../imap-data-exploration/data/phyloseq_extra_objects.rda"</span>, verbose <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<pre><code>Loading objects:
  psextra_clr_dietswap
  psextra_id_dietswap
  psextra_log10p_dietswap</code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"../imap-data-exploration/data/ps_transformed.rda"</span>, verbose <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<pre><code>Loading objects:
  ps_asin
  ps_identity
  ps_compositional
  ps_z_otu
  ps_z_sample
  ps_log10
  ps_log10p
  ps_clr
  ps_shift
  ps_scale</code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"../imap-data-exploration/data/bray_distances.rda"</span>, verbose <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<pre><code>Loading objects:
  ps_asin_bray_dist
  ps_compositional_bray_dist
  ps_z_otu_bray_dist
  ps_z_sample_bray_dist
  ps_log10_bray_dist
  ps_log10p_bray_dist
  ps_clr_bray_dist
  ps_shift_bray_dist
  ps_scale_bray_dist</code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"../imap-data-exploration/data/psextra_distances.rda"</span>, verbose <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<pre><code>Loading objects:
  psextra_clr_asin_bray_dist
  psextra_id_asin_bray_dist
  psextra_log10p_asin_bray_dist</code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"../imap-data-exploration/data/reduced_dimension.rda"</span>, verbose <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<pre><code>Loading objects:
  pca_asin_bray_metrics
  mds_asin_bray_metrics
  pcoa_asin_bray_metrics
  tsne_asin_bray_metrics</code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"../imap-data-exploration/data/phyloseq_raw_rel_psextra_df_objects.rda"</span>, verbose <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<pre><code>Loading objects:
  ps_raw
  ps_rel
  psextra_raw
  psextra_rel
  ps_df</code></pre>
<div id="permanova" class="section level2" number="3.1">
<h2>
<span class="header-section-number">3.1</span> Permutational Multivariate Analysis of Variance<a class="anchor" aria-label="anchor" href="#permanova"><i class="fas fa-link"></i></a>
</h2>
<p>PERMANOVA (Permutational Multivariate Analysis of Variance) is a statistical test used to assess the significance of differences between groups of microbial communities. Particularly suited for analyzing multivariate data like microbiome composition, PERMANOVA offers valuable insights into how different experimental conditions or treatments impact microbial community structure.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://david-barnett.github.io/microViz">microViz</a></span><span class="op">)</span> </span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span> </span>
<span></span>
<span><span class="va">bray_perm</span> <span class="op">&lt;-</span> <span class="va">psextra_log10p_asin_bray_dist</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://david-barnett.github.io/microViz/reference/dist_permanova.html">dist_permanova</a></span><span class="op">(</span></span>
<span>    seed <span class="op">=</span> <span class="fl">1234</span>, <span class="co"># for set.seed to ensure reproducibility of random process</span></span>
<span>    n_processes <span class="op">=</span> <span class="fl">1</span>, n_perms <span class="op">=</span> <span class="fl">99</span>, <span class="co"># you should use at least 999!</span></span>
<span>    variables <span class="op">=</span> <span class="st">"bmi_group"</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://david-barnett.github.io/microViz/reference/psExtra-accessors.html">perm_get</a></span><span class="op">(</span><span class="va">bray_perm</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>           Df  SumOfSqs         R2        F Pr(&gt;F)
bmi_group   2 0.1233611 0.04544245 5.212833   0.01
Residual  219 2.5913059 0.95455755       NA     NA
Total     221 2.7146670 1.00000000       NA     NA</code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://david-barnett.github.io/microViz/reference/psExtra-accessors.html">info_get</a></span><span class="op">(</span><span class="va">bray_perm</span><span class="op">)</span></span></code></pre></div>
<pre><code>psExtra info:
tax_agg = "Genus" tax_trans = "log10p" dist_method = "bray" </code></pre>
</div>
<div id="lda-effect-size" class="section level2" number="3.2">
<h2>
<span class="header-section-number">3.2</span> LEfSe: Linear discriminant analysis (LDA) effect size<a class="anchor" aria-label="anchor" href="#lda-effect-size"><i class="fas fa-link"></i></a>
</h2>
<p>Linear discriminant analysis effect size (LEfSe) is a statistical method used for biomarker discovery and identification of differentially abundant features between two or more biological groups.</p>
<ul>
<li>LEfSe combines LDA (Linear Discriminant Analysis) and effect size estimation to identify microbial taxa or features that are most likely to explain differences between groups.</li>
</ul>
<p><strong>LDA plot</strong></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files2.html">dir.create</a></span><span class="op">(</span><span class="st">"data"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/files2.html">dir.create</a></span><span class="op">(</span><span class="st">"data"</span><span class="op">)</span><span class="op">}</span></span>
<span></span>
<span><span class="co"># source("_microbial_plot.R")</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">microbial</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Nationality</span></span>
<span><span class="va">lda_nat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/microbial/man/ldamarker.html">ldamarker</a></span><span class="op">(</span><span class="va">ps_raw</span>, group<span class="op">=</span><span class="st">"nationality"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">lda_nat</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/microbial/man/plotLDA.html">plotLDA</a></span><span class="op">(</span>group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"AAM"</span>,<span class="st">"AFR"</span><span class="op">)</span>, lda <span class="op">=</span> <span class="fl">4.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># BMI</span></span>
<span><span class="va">lda_bmi</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/microbial/man/ldamarker.html">ldamarker</a></span><span class="op">(</span><span class="va">ps_raw</span>, group<span class="op">=</span><span class="st">"bmi_group"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">lda_bmi</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/microbial/man/plotLDA.html">plotLDA</a></span><span class="op">(</span>group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"lean"</span>, <span class="st">"overweight"</span><span class="op">)</span>, lda <span class="op">=</span> <span class="fl">4.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="figureslefse_LDA-1.png" width="672"></div>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lda_bmi</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/microbial/man/plotLDA.html">plotLDA</a></span><span class="op">(</span>group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"lean"</span>, <span class="st">"obese"</span><span class="op">)</span>, lda <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="figureslefse_LDA-2.png" width="672"></div>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lda_bmi</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/microbial/man/plotLDA.html">plotLDA</a></span><span class="op">(</span>group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"overweight"</span>, <span class="st">"obese"</span><span class="op">)</span>, lda <span class="op">=</span> <span class="fl">4.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="figureslefse_LDA-3.png" width="672"></div>
</div>
<div id="taxa-prevalence-and-detection" class="section level2" number="3.3">
<h2>
<span class="header-section-number">3.3</span> Taxa Prevalence and Detection<a class="anchor" aria-label="anchor" href="#taxa-prevalence-and-detection"><i class="fas fa-link"></i></a>
</h2>
<p>Taxa prevalence refers to the frequency or occurrence of a particular taxonomic unit (e.g., a species or genus) within a given dataset or set of samples.</p>
<ul>
<li>It quantifies how often a taxon is present or detected across the samples under study.</li>
<li>High prevalence indicates that the taxon is commonly found across many samples</li>
<li>Low prevalence suggests that the taxon is more sporadic or rare.</li>
<li>Analyzing taxa prevalence can provide insights into the distribution of different microbial taxa.</li>
</ul>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217">phyloseq</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Min and max sample and taxa sums: These values are useful when setting up some threshholds.</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"Minimum sample sums:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_sums.html">sample_sums</a></span><span class="op">(</span><span class="va">ps_raw</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, <span class="st">"\n\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code>Minimum sample sums: 1776 </code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"Maximum sample sums:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_sums.html">sample_sums</a></span><span class="op">(</span><span class="va">ps_raw</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, <span class="st">"\n\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code>Maximum sample sums: 28883 </code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"Minimum taxa sums:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_sums.html">taxa_sums</a></span><span class="op">(</span><span class="va">ps_raw</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, <span class="st">"\n\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code>Minimum taxa sums: 0 </code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"Maximum taxa sums:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_sums.html">taxa_sums</a></span><span class="op">(</span><span class="va">ps_raw</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, <span class="st">"\n\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code>Maximum taxa sums: 873314 </code></pre>
<p><strong>Subsetting to core taxa</strong></p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ps_prune_sample</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_samples-methods.html">subset_samples</a></span><span class="op">(</span><span class="va">ps_raw</span>, <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_sums.html">sample_sums</a></span><span class="op">(</span><span class="va">ps_raw</span><span class="op">)</span> <span class="op">&gt;</span><span class="fl">20000</span><span class="op">)</span></span>
<span><span class="va">ps_prune_taxa</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_taxa-methods.html">prune_taxa</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_sums.html">taxa_sums</a></span><span class="op">(</span><span class="va">ps_prune_sample</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="va">ps_prune_sample</span><span class="op">)</span></span>
<span><span class="va">ps_core</span> <span class="op">&lt;-</span> <span class="fu">microbiome</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microbiome/man/core.html">core</a></span><span class="op">(</span><span class="va">ps_prune_taxa</span>, detection  <span class="op">=</span> <span class="fl">0.01</span>, prevalence <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span>
<span><span class="va">ps_core</span></span></code></pre></div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 116 taxa and 18 samples ]
sample_data() Sample Data:       [ 18 samples by 8 sample variables ]
tax_table()   Taxonomy Table:    [ 116 taxa by 3 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 116 tips and 115 internal nodes ]</code></pre>
</div>
<div id="permanova" class="section level2" number="3.4">
<h2>
<span class="header-section-number">3.4</span> PERMANOVA: Permutation Analysis of Variance<a class="anchor" aria-label="anchor" href="#permanova"><i class="fas fa-link"></i></a>
</h2>
<p>Use PERMANOVA to assess the significance of differences between groups based on multivariate data. PERMANOVA is a flexible and robust method that accounts for various data characteristics and can provide valuable insights into the factors driving variation in the data.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Permanova</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://haozhu233.github.io/kableExtra/">kableExtra</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org">magrittr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">ps_prune_sample</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_samples-methods.html">subset_samples</a></span><span class="op">(</span><span class="va">ps_raw</span>, <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_sums.html">sample_sums</a></span><span class="op">(</span><span class="va">ps_raw</span><span class="op">)</span> <span class="op">&gt;</span><span class="fl">15000</span><span class="op">)</span></span>
<span><span class="va">ps_prune_taxa</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_taxa-methods.html">prune_taxa</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_sums.html">taxa_sums</a></span><span class="op">(</span><span class="va">ps_prune_sample</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="va">ps_prune_sample</span><span class="op">)</span></span>
<span><span class="va">ps_core</span> <span class="op">&lt;-</span> <span class="fu">microbiome</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microbiome/man/core.html">core</a></span><span class="op">(</span><span class="va">ps_prune_taxa</span>, detection  <span class="op">=</span> <span class="fl">0.01</span>, prevalence <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span>
<span><span class="va">ps_core</span></span></code></pre></div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 119 taxa and 76 samples ]
sample_data() Sample Data:       [ 76 samples by 8 sample variables ]
tax_table()   Taxonomy Table:    [ 119 taxa by 3 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 119 tips and 118 internal nodes ]</code></pre>
</div>
<div id="population-wise-multivariate-anova" class="section level2" number="3.5">
<h2>
<span class="header-section-number">3.5</span> Population-wise multivariate ANOVA<a class="anchor" aria-label="anchor" href="#population-wise-multivariate-anova"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vegandevs/vegan">vegan</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">microbial</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ps_raw</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span><span class="fu"><a href="https://rdrr.io/pkg/microbial/man/betatest.html">betatest</a></span><span class="op">(</span>group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"nationality"</span>, <span class="st">"bmi_group"</span><span class="op">)</span>, distance <span class="op">=</span> <span class="st">"bray"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://magrittr.tidyverse.org/reference/aliases.html">set_colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="st">"Df"</span>, <span class="st">"SumsOfSqs"</span>, <span class="st">"MeanSqs"</span>, <span class="st">"FModel"</span>, <span class="st">"R2"</span>, <span class="st">"Pvalue"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span>format <span class="op">=</span> <span class="st">"html"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html">kable_styling</a></span><span class="op">(</span>bootstrap_options <span class="op">=</span> <span class="st">"basic"</span>, full_width <span class="op">=</span> <span class="cn">F</span>, position <span class="op">=</span> <span class="st">"float_left"</span><span class="op">)</span>  <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/column_spec.html">column_spec</a></span><span class="op">(</span><span class="fl">7</span>, bold <span class="op">=</span> <span class="cn">TRUE</span>, color <span class="op">=</span> <span class="st">"black"</span>, background <span class="op">=</span> <span class="st">"#eeeeee"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-table"><table class="table" style="width: auto !important; float: left; margin-right: 10px;">
<thead><tr>
<th style="text-align:left;">
Group
</th>
<th style="text-align:right;">
Df
</th>
<th style="text-align:right;">
SumsOfSqs
</th>
<th style="text-align:right;">
MeanSqs
</th>
<th style="text-align:right;">
FModel
</th>
<th style="text-align:right;">
R2
</th>
<th style="text-align:right;">
Pvalue
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
bmi_group
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2.303012
</td>
<td style="text-align:right;">
1.1515062
</td>
<td style="text-align:right;">
6.793851
</td>
<td style="text-align:right;">
0.0584197
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
0.001
</td>
</tr>
<tr>
<td style="text-align:left;">
bmi_group
</td>
<td style="text-align:right;">
219
</td>
<td style="text-align:right;">
37.118834
</td>
<td style="text-align:right;">
0.1694924
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.9415803
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
bmi_group
</td>
<td style="text-align:right;">
221
</td>
<td style="text-align:right;">
39.421847
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
nationality
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
6.940786
</td>
<td style="text-align:right;">
6.9407856
</td>
<td style="text-align:right;">
47.011174
</td>
<td style="text-align:right;">
0.1760644
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
0.001
</td>
</tr>
<tr>
<td style="text-align:left;">
nationality
</td>
<td style="text-align:right;">
220
</td>
<td style="text-align:right;">
32.481061
</td>
<td style="text-align:right;">
0.1476412
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.8239356
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
nationality
</td>
<td style="text-align:right;">
221
</td>
<td style="text-align:right;">
39.421847
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
NA
</td>
</tr>
</tbody>
</table></div>
</div>
<div id="core-taxa-multivariate-anova" class="section level2" number="3.6">
<h2>
<span class="header-section-number">3.6</span> Core taxa multivariate ANOVA<a class="anchor" aria-label="anchor" href="#core-taxa-multivariate-anova"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ps_core</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span><span class="fu"><a href="https://rdrr.io/pkg/microbial/man/betatest.html">betatest</a></span><span class="op">(</span>group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"nationality"</span>, <span class="st">"bmi_group"</span><span class="op">)</span>, distance <span class="op">=</span> <span class="st">"bray"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://magrittr.tidyverse.org/reference/aliases.html">set_colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="st">"Df"</span>, <span class="st">"SumsOfSqs"</span>, <span class="st">"MeanSqs"</span>, <span class="st">"FModel"</span>, <span class="st">"R2"</span>, <span class="st">"Pvalue"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span>format <span class="op">=</span> <span class="st">"html"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html">kable_styling</a></span><span class="op">(</span>bootstrap_options <span class="op">=</span> <span class="st">"basic"</span>, full_width <span class="op">=</span> <span class="cn">F</span>, position <span class="op">=</span> <span class="st">"float_left"</span><span class="op">)</span>  <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/column_spec.html">column_spec</a></span><span class="op">(</span><span class="fl">7</span>, bold <span class="op">=</span> <span class="cn">TRUE</span>, color <span class="op">=</span> <span class="st">"black"</span>, background <span class="op">=</span> <span class="st">"#eeeeee"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-table"><table class="table" style="width: auto !important; float: left; margin-right: 10px;">
<thead><tr>
<th style="text-align:left;">
Group
</th>
<th style="text-align:right;">
Df
</th>
<th style="text-align:right;">
SumsOfSqs
</th>
<th style="text-align:right;">
MeanSqs
</th>
<th style="text-align:right;">
FModel
</th>
<th style="text-align:right;">
R2
</th>
<th style="text-align:right;">
Pvalue
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
bmi_group
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
0.3580195
</td>
<td style="text-align:right;">
0.1790097
</td>
<td style="text-align:right;">
1.405949
</td>
<td style="text-align:right;">
0.0370904
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
0.196
</td>
</tr>
<tr>
<td style="text-align:left;">
bmi_group
</td>
<td style="text-align:right;">
73
</td>
<td style="text-align:right;">
9.2945867
</td>
<td style="text-align:right;">
0.1273231
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.9629096
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
bmi_group
</td>
<td style="text-align:right;">
75
</td>
<td style="text-align:right;">
9.6526062
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
nationality
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1.4483886
</td>
<td style="text-align:right;">
1.4483886
</td>
<td style="text-align:right;">
13.064104
</td>
<td style="text-align:right;">
0.1500516
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
0.001
</td>
</tr>
<tr>
<td style="text-align:left;">
nationality
</td>
<td style="text-align:right;">
74
</td>
<td style="text-align:right;">
8.2042176
</td>
<td style="text-align:right;">
0.1108678
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.8499484
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
nationality
</td>
<td style="text-align:right;">
75
</td>
<td style="text-align:right;">
9.6526062
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;font-weight: bold;color: black !important;background-color: rgba(238, 238, 238, 255) !important;">
NA
</td>
</tr>
</tbody>
</table></div>
</div>
<div id="diff_abund" class="section level2" number="3.7">
<h2>
<span class="header-section-number">3.7</span> Differential Abundance Analysis with the difftest() Function<a class="anchor" aria-label="anchor" href="#diff_abund"><i class="fas fa-link"></i></a>
</h2>
<p>The <code><a href="https://rdrr.io/pkg/microbial/man/difftest.html">difftest()</a></code> function from the microbial package serves as a robust solution for conducting differential abundance testing in microbiome data analysis. Its core objective is to identify taxa (such as bacteria and fungi) that demonstrate noteworthy differences in abundance levels across two or more groups of samples.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/ggtext/">ggtext</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">microbial</span><span class="op">)</span></span>
<span></span>
<span><span class="va">difftest</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/microbial/man/difftest.html">difftest</a></span><span class="op">(</span><span class="va">ps_raw</span>, group<span class="op">=</span><span class="st">"nationality"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">difftest</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">log2FoldChange</span>, <span class="va">pvalue</span>, <span class="va">padj</span>, <span class="va">Significant</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">Significant</span> <span class="op">==</span> <span class="st">"Yes"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kbl.html">kbl</a></span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"Differential abundance results"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html">kable_styling</a></span><span class="op">(</span>bootstrap_options <span class="op">=</span> <span class="st">"basic"</span>, full_width <span class="op">=</span> <span class="cn">F</span>, position <span class="op">=</span> <span class="st">"float_left"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-table"><table class="table" style="width: auto !important; float: left; margin-right: 10px;">
<caption>
(#tab:diffabund_table)Differential abundance results
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
log2FoldChange
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
padj
</th>
<th style="text-align:left;">
Significant
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
OTU016
</td>
<td style="text-align:right;">
-0.4587101
</td>
<td style="text-align:right;">
0.0043914
</td>
<td style="text-align:right;">
0.0084994
</td>
<td style="text-align:left;">
Yes
</td>
</tr>
<tr>
<td style="text-align:left;">
OTU055
</td>
<td style="text-align:right;">
0.4486681
</td>
<td style="text-align:right;">
0.0009475
</td>
<td style="text-align:right;">
0.0021056
</td>
<td style="text-align:left;">
Yes
</td>
</tr>
<tr>
<td style="text-align:left;">
OTU095
</td>
<td style="text-align:right;">
-1.4494599
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:left;">
Yes
</td>
</tr>
<tr>
<td style="text-align:left;">
OTU044
</td>
<td style="text-align:right;">
-1.4659327
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:left;">
Yes
</td>
</tr>
<tr>
<td style="text-align:left;">
OTU097
</td>
<td style="text-align:right;">
0.3855880
</td>
<td style="text-align:right;">
0.0200319
</td>
<td style="text-align:right;">
0.0333865
</td>
<td style="text-align:left;">
Yes
</td>
</tr>
<tr>
<td style="text-align:left;">
OTU033
</td>
<td style="text-align:right;">
-0.6708167
</td>
<td style="text-align:right;">
0.0001874
</td>
<td style="text-align:right;">
0.0004409
</td>
<td style="text-align:left;">
Yes
</td>
</tr>
<tr>
<td style="text-align:left;">
OTU092
</td>
<td style="text-align:right;">
-1.0797983
</td>
<td style="text-align:right;">
0.0000003
</td>
<td style="text-align:right;">
0.0000008
</td>
<td style="text-align:left;">
Yes
</td>
</tr>
<tr>
<td style="text-align:left;">
OTU085
</td>
<td style="text-align:right;">
-0.3184212
</td>
<td style="text-align:right;">
0.0013021
</td>
<td style="text-align:right;">
0.0027714
</td>
<td style="text-align:left;">
Yes
</td>
</tr>
<tr>
<td style="text-align:left;">
OTU023
</td>
<td style="text-align:right;">
4.0345186
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:left;">
Yes
</td>
</tr>
<tr>
<td style="text-align:left;">
OTU018
</td>
<td style="text-align:right;">
0.7328485
</td>
<td style="text-align:right;">
0.0000001
</td>
<td style="text-align:right;">
0.0000005
</td>
<td style="text-align:left;">
Yes
</td>
</tr>
</tbody>
</table></div>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Differential abundance plot</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">microbial</span><span class="op">)</span></span>
<span></span>
<span><span class="va">difftest_w_meta</span> <span class="op">&lt;-</span> <span class="va">difftest</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select_all.html">rename_all</a></span><span class="op">(</span><span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/make.unique.html">make.unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/chartr.html">tolower</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"otu"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">inner_join</a></span><span class="op">(</span><span class="va">.</span>, <span class="fu">psmelt</span><span class="op">(</span><span class="va">ps_raw</span><span class="op">)</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"otu"</span> <span class="op">=</span> <span class="st">"OTU"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/relocate.html">relocate</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">nationality</span>, <span class="va">bmi_group</span><span class="op">)</span>, .after <span class="op">=</span> <span class="va">otu</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>nationality <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">nationality</span>,</span>
<span>                      levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"AAM"</span>, <span class="st">"AFR"</span><span class="op">)</span>,</span>
<span>                      labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"African American"</span>, <span class="st">"African"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         bmi_group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">bmi_group</span>,</span>
<span>                      levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"lean"</span>, <span class="st">"overweight"</span>, <span class="st">"obese"</span><span class="op">)</span>,</span>
<span>                      labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Lean"</span>, <span class="st">"Overweight"</span>, <span class="st">"Obese"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select_all.html">rename_all</a></span><span class="op">(</span><span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/make.unique.html">make.unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/chartr.html">tolower</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">phylum_1</span>, <span class="op">-</span><span class="va">family_1</span>, <span class="op">-</span><span class="va">genus_1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="va">otu</span>, .keep_all <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"phylum"</span>, <span class="st">"family"</span>, <span class="st">"genus"</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"level"</span>, values_to <span class="op">=</span> <span class="st">"taxon"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>taxon <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace</a></span><span class="op">(</span>string <span class="op">=</span> <span class="va">taxon</span>,</span>
<span>                            pattern <span class="op">=</span> <span class="st">"(.*)"</span>,</span>
<span>                            replacement <span class="op">=</span> <span class="st">"*\\1*"</span><span class="op">)</span>,</span>
<span>        taxon <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace</a></span><span class="op">(</span>string <span class="op">=</span> <span class="va">taxon</span>,</span>
<span>                            pattern <span class="op">=</span> <span class="st">"\\*(.*)_unclassified\\*"</span>,</span>
<span>                            replacement <span class="op">=</span> <span class="st">"Unclassified&lt;br&gt;*\\1*"</span><span class="op">)</span>,</span>
<span>        taxon <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace_all</a></span><span class="op">(</span><span class="va">taxon</span>, <span class="st">"_"</span>, <span class="st">" "</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.0001</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">level</span> <span class="op">==</span> <span class="st">"genus"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">difftest_w_meta</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/reorder.factor.html">reorder</a></span><span class="op">(</span><span class="va">taxon</span>, <span class="va">log2foldchange</span><span class="op">)</span>, x <span class="op">=</span> <span class="va">log2foldchange</span>, fill <span class="op">=</span> <span class="va">nationality</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_col</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_light</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Log2 Fold Change"</span>, y <span class="op">=</span> <span class="st">""</span>, fill <span class="op">=</span> <span class="cn">NULL</span>, subtitle <span class="op">=</span> <span class="st">"Differential abundance by nationality"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text.y <span class="op">=</span> <span class="fu"><a href="https://wilkelab.org/ggtext/reference/element_markdown.html">element_markdown</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">7</span><span class="op">)</span>,</span>
<span>        legend.text <span class="op">=</span> <span class="fu"><a href="https://wilkelab.org/ggtext/reference/element_markdown.html">element_markdown</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">7</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html">coord_cartesian</a></span><span class="op">(</span>xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">3</span>, <span class="fl">4</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="figuresdiffabund_table-1.png" width="672"></div>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">difftest_w_meta</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/reorder.factor.html">reorder</a></span><span class="op">(</span><span class="va">taxon</span>, <span class="va">log2foldchange</span><span class="op">)</span>, x <span class="op">=</span> <span class="va">log2foldchange</span>, fill <span class="op">=</span> <span class="va">bmi_group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_col</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_light</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Log2 Fold Change"</span>, y <span class="op">=</span> <span class="st">""</span>, fill <span class="op">=</span> <span class="cn">NULL</span>, subtitle <span class="op">=</span> <span class="st">"Differential abundance by BMI"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text.y <span class="op">=</span> <span class="fu"><a href="https://wilkelab.org/ggtext/reference/element_markdown.html">element_markdown</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">7</span><span class="op">)</span>,</span>
<span>        legend.text <span class="op">=</span> <span class="fu"><a href="https://wilkelab.org/ggtext/reference/element_markdown.html">element_markdown</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">7</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html">coord_cartesian</a></span><span class="op">(</span>xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">3</span>, <span class="fl">4</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="figuresdiffabund_table-2.png" width="672"></div>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/save.html">save</a></span><span class="op">(</span><span class="va">difftest_w_meta</span>, file <span class="op">=</span> <span class="st">"data/difftest_w_meta.rda"</span><span class="op">)</span></span></code></pre></div>
</div>
<div id="biomarker-discovery" class="section level2" number="3.8">
<h2>
<span class="header-section-number">3.8</span> Biomarker Discovery<a class="anchor" aria-label="anchor" href="#biomarker-discovery"><i class="fas fa-link"></i></a>
</h2>
<p>Biomarker discovery aims to identify distinct markers or indicators that offer valuable insights into health and disease. These identified biomarkers provide crucial information regarding various aspects. Employing machine learning techniques, such as <code>random forest classification</code>, can further enhance the understanding by providing more in-depth insights, including:</p>
<ul>
<li>Facilitating disease diagnosis</li>
<li>Monitoring and evaluating treatment responses</li>
<li>Improving comprehension of underlying biological processes</li>
<li>Enabling the prediction of outcomes.</li>
</ul>
<div id="microbial-pkg-markers" class="section level3" number="3.8.1">
<h3>
<span class="header-section-number">3.8.1</span> Biomarkers using microbial package<a class="anchor" aria-label="anchor" href="#microbial-pkg-markers"><i class="fas fa-link"></i></a>
</h3>
<p>Microbial R package performms Random Forest Classification then computes the Confusion Matrix as a means of evaluating the model’s performance.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">microbial</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/microbial/man/biomarker.html">biomarker</a></span><span class="op">(</span><span class="va">ps_raw</span>, group <span class="op">=</span> <span class="st">"nationality"</span>, ntree <span class="op">=</span> <span class="fl">129</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/microbial/man/plotmarker.html">plotmarker</a></span><span class="op">(</span>level<span class="op">=</span><span class="st">"Genus"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_test</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<pre><code>
Call:
 randomForest(formula = group ~ ., data = data, importance = TRUE,      proximity = TRUE, ntree = ntree) 
               Type of random forest: classification
                     Number of trees: 129
No. of variables tried at each split: 8

        OOB estimate of  error rate: 4.5%
Confusion matrix:
    AAM AFR class.error
AAM 119   4  0.03252033
AFR   6  93  0.06060606</code></pre>
<div class="inline-figure"><img src="figuresbiomarker_plot-1.png" width="576"></div>
<blockquote>
<p>More on biomarker discovery techniques in the next sections underMicrobial Machine Learning.</p>
</blockquote>

</div>
</div>
</div>



  <div class="chapter-nav">
<div class="prev"><a href="microbiome-statistical-analysis-steps.html"><span class="header-section-number">1</span> Microbiome Statistical Analysis Steps</a></div>
<div class="next"><a href="imap-github-repos.html"><span class="header-section-number">A</span> IMAP GitHub Repos</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#statistical-significance-analysis"><span class="header-section-number">3</span> Statistical Significance Analysis</a></li>
<li><a class="nav-link" href="#permanova"><span class="header-section-number">3.1</span> Permutational Multivariate Analysis of Variance</a></li>
<li><a class="nav-link" href="#lda-effect-size"><span class="header-section-number">3.2</span> LEfSe: Linear discriminant analysis (LDA) effect size</a></li>
<li><a class="nav-link" href="#taxa-prevalence-and-detection"><span class="header-section-number">3.3</span> Taxa Prevalence and Detection</a></li>
<li><a class="nav-link" href="#permanova"><span class="header-section-number">3.4</span> PERMANOVA: Permutation Analysis of Variance</a></li>
<li><a class="nav-link" href="#population-wise-multivariate-anova"><span class="header-section-number">3.5</span> Population-wise multivariate ANOVA</a></li>
<li><a class="nav-link" href="#core-taxa-multivariate-anova"><span class="header-section-number">3.6</span> Core taxa multivariate ANOVA</a></li>
<li><a class="nav-link" href="#diff_abund"><span class="header-section-number">3.7</span> Differential Abundance Analysis with the difftest() Function</a></li>
<li>
<a class="nav-link" href="#biomarker-discovery"><span class="header-section-number">3.8</span> Biomarker Discovery</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#microbial-pkg-markers"><span class="header-section-number">3.8.1</span> Biomarkers using microbial package</a></li></ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/tmbuza/imap-machine-learning/blob/gh-pages/01_statistical_analysis.Rmd">View source <i class="fas fa-air-freshener"></i></a></li>
          <li><a id="book-edit" href="https://github.com/tmbuza/imap-machine-learning/edit/gh-pages/01_statistical_analysis.Rmd">Edit this page <i class="fas fa-air-freshener"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container">
    <div class="row">
      <div class="col-12 col-md-6 mt-3">
        <p><strong><u><a class="text-light" href="https://microbiome.complexdatainsights.com"><big>IMAP:</big> Integrated Microbiome Analysis Pipelines</a></u></strong>: End-to-End Practical User Guides Using Integrated Approaches</p>
<br><p>Authored by Teresia Mrema-Buza, Last updated: <b>2024-04-18</b>.</p>
<br>
</div>
      <div class="col-12 col-md-6 mt-3">
        <p>Note: This practical guide was created using the <a class="text-light" href="https://bookdown.org/yihui/bookdown/">bookdown</a> R package, styled with the modern bs4_book format.</p>
      </div>

    </div>
  </div>
</footer>
</body>
</html>
